lib/dk_buttons_and_leds/dk_buttons_and_leds.c

```diff
@@ -63,8 +67,13 @@ static int callback_ctrl(bool enable)
 	 */
 	for (size_t i = 0; (i < ARRAY_SIZE(buttons)) && !err; i++) {
 		if (enable) {
-			flags = ((buttons[i].port == GPIO0_DEV) ? FLAGS_GPIO0_ACTIVE :
-								  FLAGS_GPIO_EXP_ACTIVE);
+			flags = ((buttons[i].port == GPIO0_DEV ||
+#if DT_NODE_EXISTS(DT_NODELABEL(gpio1))
+				  buttons[i].port == GPIO1_DEV)
+#endif
+					 ?
+					 FLAGS_GPIO_0_1_ACTIVE :
+					 FLAGS_GPIO_EXP_ACTIVE);
```

Comment:
I think this might clean it up a bit (together with suggestion above):
```suggestion
		if (enable && (buttons[i].port == GPIO0_DEV || buttons[i].port == GPIO1_DEV)) {
			flags = FLAGS_GPIO_0_1_ACTIVE;
		} else if (enable) {
			/* External GPIO */
			flags = FLAGS_GPIO_EXP_ACTIVE;
		} else {
			flags = GPIO_INT_DISABLE;
		}
```